MediaBrowser.Providers/Lyric/LrcLyricProvider.cs

```diff
@@ -0,0 +1,213 @@
+    public LrcLyricProvider(ILogger<LrcLyricProvider> logger)
+    {
+        _logger = logger;
+    }
+
+    /// <inheritdoc />
+    public string Name => "LrcLyricProvider";
+
+    /// <summary>
+    /// Gets the priority.
+    /// </summary>
+    /// <value>The priority.</value>
+    public ResolverPriority Priority => ResolverPriority.First;
+
+    /// <inheritdoc />
+    public IReadOnlyCollection<string> SupportedMediaTypes { get; } = new[] { "lrc" };
+
+    /// <summary>
+    /// Opens lyric file for the requested item, and processes it for API return.
+    /// </summary>
+    /// <param name="item">The item to to process.</param>
+    /// <returns>If provider can determine lyrics, returns a <see cref="LyricResponse"/> with or without metadata; otherwise, null.</returns>
+    public LyricResponse? GetLyrics(BaseItem item)
+    {
+        string? lyricFilePath = LyricInfo.GetLyricFilePath(this, item.Path);
+
+        if (string.IsNullOrEmpty(lyricFilePath))
+        {
+            return null;
+        }
+
+        List<LyricLine> lyricList = new List<LyricLine>();
+        List<LrcParser.Model.Lyric> sortedLyricData = new List<LrcParser.Model.Lyric>();
+
+        IDictionary<string, string> fileMetaData = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+        string lrcFileContent = System.IO.File.ReadAllText(lyricFilePath);
+
+        try
+        {
+            // Parse and sort lyric rows
+            LyricParser lrcLyricParser = new LrcParser.Parser.Lrc.LrcParser();
+            Song lyricData = lrcLyricParser.Decode(lrcFileContent);
+            sortedLyricData = lyricData.Lyrics.Where(x => x.TimeTags.Count > 0).OrderBy(x => x.TimeTags.First().Value).ToList();
+
+            // Parse metadata rows
+            var metaDataRows = lyricData.Lyrics
+                .Where(x => x.TimeTags.Count == 0)
+                .Where(x => x.Text.StartsWith('[') && x.Text.EndsWith(']'))
+                .Select(x => x.Text)
+                .ToList();
+
+            foreach (string metaDataRow in metaDataRows)
+            {
+                int colonCount = metaDataRow.Count(f => (f == ':'));
+                if (colonCount == 0)
+                {
+                    continue;
+                }
+
+                string[] metaDataField;
+                string metaDataFieldName;
+                string metaDataFieldValue;
+
+                if (colonCount == 1)
+                {
+                    metaDataField = metaDataRow.Split(':');
+                    metaDataFieldName = metaDataField[0][1..].Trim();
+                    metaDataFieldValue = metaDataField[1][..^1].Trim();
+                }
+                else
+                {
+                    int colonIndex = metaDataRow.IndexOf(':', StringComparison.OrdinalIgnoreCase);
+                    metaDataFieldName = metaDataRow[..colonIndex][1..].Trim();
+                    metaDataFieldValue = metaDataRow[(colonIndex + 1)..][..^1].Trim();
+                }
```

Comment:
I don't really understand this bit. Is it equivalent to `metaDataRow.Split(':', RemoveEmptyEntries | TrimEntries)`?